fe/fe-core/src/main/java/com/starrocks/connector/PartitionUtil.java

```diff
@@ -353,45 +367,81 @@ public static Set<String> getMVPartitionName(Table table, Column partitionColumn
         if (isListPartition) {
             return Sets.newHashSet(getMVPartitionNameWithList(table, partitionColumn, partitionNames).keySet());
         } else {
-            return Sets.newHashSet(getMVPartitionNameWithRange(table, partitionColumn, partitionNames).keySet());
+            return Sets.newHashSet(getRangePartitionMapOfExternalTable(table, partitionColumn, partitionNames).keySet());
         }
     }
 
-    // Map partition values to partition ranges, eg
-    // [NULL,1992-01-01,1992-01-02,1992-01-03]
-    //               ||
-    //               \/
-    // [0000-01-01, 1992-01-01),[1992-01-01, 1992-01-02),[1992-01-02, 1992-01-03),[1993-01-03, MAX_VALUE)
-    public static Map<String, Range<PartitionKey>> getMVPartitionNameWithRange(Table table,
-                                                                               Column partitionColumn,
-                                                                               List<String> partitionNames)
+    /**
+     *  NOTE:
+     *  External tables may contain multi partition columns, so the same mv partition name may contain multi external
+     *  table partitions. eg:
+     *   partitionName1 : par_col=0/par_date=2020-01-01 => p20200101
+     *   partitionName2 : par_col=1/par_date=2020-01-01 => p20200101
+     */
+    public static Map<String, Set<String>> getMVPartitionNameMapOfExternalTable(Table table,
+                                                                                Column partitionColumn,
+                                                                                List<String> partitionNames)
             throws AnalysisException {
-        Map<String, Range<PartitionKey>> partitionRangeMap = new LinkedHashMap<>();
         List<Column> partitionColumns = getPartitionColumns(table);
-
         // Get the index of partitionColumn when table has multi partition columns.
         int partitionColumnIndex = checkAndGetPartitionColumnIndex(partitionColumns, partitionColumn);
+        Map<String, Set<String>> mvPartitionKeySetMap = Maps.newHashMap();
+        for (String partitionName : partitionNames) {
+            List<String> partitionNameValues = toPartitionValues(partitionName);
+            PartitionKey partitionKey = createPartitionKey(
+                    ImmutableList.of(partitionNameValues.get(partitionColumnIndex)),
+                    ImmutableList.of(partitionColumns.get(partitionColumnIndex)),
+                    table.getType());
+            String mvPartitionName = generateMVPartitionName(partitionKey);
+            mvPartitionKeySetMap.computeIfAbsent(mvPartitionName, x -> Sets.newHashSet())
+                    .add(partitionName);
+        }
+        return mvPartitionKeySetMap;
```

Comment:
Could you please add a MetaFunction to inspect this partition mapping  ?

Reply:
Good advise~  I can add one function later.

In this pr, I have add  more  refresh partition info in `[MVTaskRunExtraMessage.java](https://github.com/StarRocks/starrocks/pull/28093/files/2d42190576e0d32d097f87352699e29d5d3f74cf#diff-2beea2501608af783672f127f5dc864766939d48a820104b4f71f0bb4440c371)`